snuba/consumers/utils.py

```diff
@@ -0,0 +1,54 @@
+import logging
+import time
+
+from confluent_kafka import KafkaException
+from confluent_kafka.admin import AdminClient
+
+from snuba.utils.streams.configuration_builder import get_default_kafka_configuration
+from snuba.utils.streams.topics import Topic
+
+logger = logging.getLogger("snuba.consumer.utils")
+
+
+class InvalidTopicName(Exception):
+    pass
+
+
+def get_partition_count(topic: Topic) -> int:
+    override_params = {
+        # Default is 60s, do we need that long?
+        "socket.timeout.ms": 2000,
+    }
+    attempts = 0
+    while True:
+        try:
+            logger.info("Attempting to connect to Kafka (attempt %d)...", attempts)
+            client = AdminClient(
+                get_default_kafka_configuration(
+                    topic=topic, override_params=override_params
+                )
+            )
+            cluster_metadata = client.list_topics(topic=topic.value, timeout=2.0)
+            break
+        except KafkaException as err:
+            logger.debug(
+                "Connection to Kafka failed (attempt %d)", attempts, exc_info=err
+            )
+            attempts += 1
+            # How many attempts is too many?
+            if attempts == 3:
+                raise
+            time.sleep(1)
+
+    logger.info(f"Checking topic metadata for {topic.value}...")
```

Comment:
I think you may want to move this line above the loop, so that, if the consumer hangs, you know what it was trying to do.